src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -79,5 +83,197 @@ public string VersionString
                 return _versionString;
             }
         }
+
+        /// <summary>
+        /// Indicates whether the current application is running on the specified platform.
+        /// </summary>
+        /// <param name="platform">Case-insensitive platform name. Examples: Browser, Linux, FreeBSD, Android, iOS, macOS, tvOS, watchOS, Windows.</param>
+        public static bool IsOSPlatform(string platform)
+        {
+            if (platform == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.platform);
+            }
+            if (platform.Length == 0)
+            {
+                ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_EmptyOrNullString, ExceptionArgument.platform);
+            }
+
+#if TARGET_BROWSER
+            return platform.Equals("BROWSER", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_WINDOWS
+            return platform.Equals("WINDOWS", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_OSX
+            return platform.Equals("OSX", StringComparison.OrdinalIgnoreCase) || platform.Equals("MACOS", StringComparison.OrdinalIgnoreCase);
+#elif TARGET_UNIX
+            return platform.Equals(s_osPlatformName ??= Interop.Sys.GetUnixName(), StringComparison.OrdinalIgnoreCase);
+#else
+#error Unknown OS
+#endif
+        }
+
+        /// <summary>
+        /// Check for the OS with a >= version comparison. Used to guard APIs that were added in the given OS release.
+        /// </summary>
+        /// <param name="platform">Case-insensitive platform name. Examples: Browser, Linux, FreeBSD, Android, iOS, macOS, tvOS, watchOS, Windows.</param>
+        /// <param name="major">Major OS version number.</param>
+        /// <param name="minor">Minor OS version number (optional).</param>
+        /// <param name="build">Build OS version number (optional).</param>
+        /// <param name="revision">Revision OS version number (optional).</param>
+        public static bool IsOSPlatformVersionAtLeast(string platform, int major, int minor = 0, int build = 0, int revision = 0)
+            => IsOSPlatform(platform) && IsOSVersionAtLeast(major, minor, build, revision);
+
+        /// <summary>
+        /// Indicates whether the current application is running as WASM in a Browser.
+        /// </summary>
+        public static bool IsBrowser() =>
+#if TARGET_BROWSER
+            true;
+#else
+            false;
+#endif
+
+        /// <summary>
+        /// Indicates whether the current application is running on Linux.
+        /// </summary>
+        public static bool IsLinux() =>
+#if TARGET_LINUX
```

Comment:
I do not think we have `TARGET_LINUX` defined in CoreLib .csproj. You will need to add it.

Same for number of others.